PhysicsTools/PyTorch/test/BuildFile.xml

```diff
@@ -1,36 +1,180 @@
+  <use name="alpaka"/>
+  <use name="cppunit"/>
+  <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="eigen"/>
+  <use name="DataFormats/Portable"/>
+  <use name="DataFormats/PortableTestObjects"/>
+  <use name="FWCore/Framework"/>
+  <use name="FWCore/ParameterSet"/>
+  <use name="FWCore/Utilities"/>
+  <use name="HeterogeneousCore/AlpakaCore"/>
+  <use name="HeterogeneousCore/AlpakaInterface"/>
+  <flags ALPAKA_BACKENDS="serial"/>
+  <flags ALPAKA_BACKENDS="cuda"/>
+</bin>
+
+<bin file="alpaka/testSOADataTypes.dev.cc testRunner.cc" name="testSOADataTypes">
+  <use name="alpaka"/>
+  <use name="cppunit"/>
+  <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="eigen"/>
+  <use name="DataFormats/Portable"/>
+  <use name="DataFormats/PortableTestObjects"/>
+  <use name="FWCore/Framework"/>
+  <use name="FWCore/ParameterSet"/>
+  <use name="FWCore/Utilities"/>
+  <use name="HeterogeneousCore/AlpakaCore"/>
+  <use name="HeterogeneousCore/AlpakaInterface"/>
+  <flags ALPAKA_BACKENDS="cuda"/>
+  <flags ALPAKA_BACKENDS="serial"/>
+</bin>
+
+<!-- <bin file="alpaka/testSOAtoTorch.dev.cc testRunner.cc" name="testSOAtoTorch">
+  <use name="alpaka"/>
+  <use name="cppunit"/>
+  <use name="eigen"/>
+  <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="FWCore/Utilities"/>
+  <flags ALPAKA_BACKENDS="cuda"/>
+  <flags ALPAKA_BACKENDS="serial"/>
+</bin> -->
+
+<bin file="alpaka/testTorchDeviceMatchAlpaka.cc testRunner.cc" name="testTorchDeviceMatchAlpaka">
+  <use name="cppunit"/>
+  <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="FWCore/Utilities"/>
+  <flags ALPAKA_BACKENDS="serial"/>
+  <flags ALPAKA_BACKENDS="cuda"/>
+</bin>
+
+<bin file="alpaka/testTorchExecutionControlGuard.cc testRunner.cc" name="testTorchExecutionControlGuard">
+  <use name="cppunit"/>
+  <use name="eigen"/>
+  <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="FWCore/Utilities"/>
+  <flags ALPAKA_BACKENDS="serial"/>
+  <flags ALPAKA_BACKENDS="cuda"/>
+</bin>
+
+<bin file="testModelCpu.cc testRunner.cc" name="testModelCpu">
+  <use name="cppunit"/>
+  <use name="pytorch"/>
+</bin>
 
+<bin file="testModelCuda.cc testRunner.cc" name="testModelCuda">
+  <use name="cppunit"/>
+  <use name="cuda"/>
+  <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="HeterogeneousCore/CUDAUtilities"/>
+</bin>
 
-<bin name="testTorch" file="testTorch.cc">
+<iftool name="cuda-gcc-support">
+<bin name="testTensorStride" file="testRunner.cc,testTensorStride.cu">
+  <use name="catch2"/>
+  <use name="cppunit"/>
+  <use name="cuda"/>
   <use name="pytorch"/>
+  <use name="pytorch-cuda"/>
+  <use name="FWCore/ParameterSet"/>
+  <use name="FWCore/ParameterSetReader"/>
+  <use name="FWCore/PluginManager"/>
+  <use name="FWCore/ServiceRegistry"/>
+  <use name="FWCore/Utilities"/>
+  <use name="HeterogeneousCore/CUDAServices"/>
+  <use name="HeterogeneousCore/CUDAUtilities"/>
```

Comment:
Of these, only `CUDAUtilities` seem to be used
```suggestion
  <use name="HeterogeneousCore/CUDAUtilities"/>
```

(same for the other test executables below)